WordPress/gutenberg
# Page list: Fix submenu colors

## What?

Adds the overlay color context to submenus in the page list block when the page list is inside the navigation block.

## Why?

Submenus in page list did not inherit the submenu and overlay colors from the navigation block. In the editor, the submenu colors were white background and black text.

## How?

Adds the has-text-color and has-background CSS classes, the palette color classes, and inline custom colors to the submenu items (the inner `<li> `element).

## Testing Instructions

Create a new page on your WordPress install.
Add a child page to this page, and a child page to that child page.
You will use these as a page list with a two level sub menu.

Create a new post or page and add a navigation block.
Add the page list to the navigation block.
In the color setting, select a custom color using the color picker for the Submenu and overlay text color, and one for the submenu and overlay background color.
Confirm that the colors are used for the submenu in the editor and front.

Return to the editor and select overlay text and background color from the palette.
Confirm that the colors are used for the submenu in the editor and front.

## Screenshots or screencast 
After:
<img width="1355" alt="pagelist submenu colors are showing in the editor" src="https://user-images.githubusercontent.com/7422055/191425980-30bf4ae7-f36c-4653-92fa-ef71ec62933c.png">
